File: solution/2485. Find the Pivot Integer/2485_01.cpp
class Solution {
public:
    int pivotInteger(int n) {
        int l = 1, r = n, mid;
        while (l <= r) {
            mid = l + (r - l) / 2;
            int sumL = mid * (1 + mid) / 2;
            int sumR = (n - mid + 1) * (mid + n) / 2;
            if (sumL == sumR)
                return mid;
            if (sumL < sumR)
                l = mid + 1;
            else
                r = mid - 1;
        }
        return -1;
    }
};

File: solution/2485. Find the Pivot Integer/2485_01.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        total = n * (1 + n) / 2
        l, r = 1, n
        while l <= r:
            mid = (l + r) // 2
            sum_l = mid * (1 + mid) / 2
            sum_r = total - sum_l + mid
            if sum_l == sum_r:
                return mid
            if sum_l < sum_r:
                l = mid + 1
            else:
                r = mid - 1
        return -1
    
